Core/clim-basic/clipboard.lisp

```diff
@@ -0,0 +1,76 @@
+(in-package :climi)
+
+;;;
+;;;  Functions dealing with copying to the clipboard.
+;;;
+
+(deftype representation-type-name () '(member :string :html :image))
+
+(defgeneric copy-to-clipboard-with-port (port sheet clipboard-p object presentation-type)
+  (:documentation "Method to be implemented by backends."))
+
+(defmethod copy-to-clipboard-with-port ((port clim:port) (sheet clim:sheet) clipboard-p object presentation-type)
+  "Fallback implementation if the implementation does not support a clipboard."
+  nil)
+
+(defun copy-to-clipboard (sheet object &key presentation-type)
+  "Copy OBJECT to the clipboard.
+SHEET is the owner of the clipboard, and it is not guaranteed that the
+content of the clipboard will be available after the sheet has been
+removed."
+  (copy-to-clipboard-with-port (port sheet) sheet t object presentation-type))
+
+(defun copy-to-selection (sheet object &key presentation-type)
+  "Copy OBJECT to the selection.
+SHEET is the owner of the selection, and it is not guaranteed that the
+content of the selection will be available after the sheet has been
+removed."
+  (copy-to-clipboard-with-port (port sheet) sheet nil object presentation-type))
+
+(defgeneric clear-clipboard-with-port (port sheet clipboard-p))
```

Comment:
please add documentation option to this function.

Reply:
Done.